src/ngraph/pass/constant_folding.cpp

```diff
@@ -763,3 +766,190 @@ void pass::ConstantFolding::construct_constant_quantize()
+        break;
+    case element::Type_t::dynamic:
+        NGRAPH_CHECK(false, "Encountered 'dynamic' element type in fold_constant_convert");
+        break;
+    case element::Type_t::boolean:
+        return fold_constant_convert_helper0<char>(constant, output_element_type);
+    case element::Type_t::bf16:
+        return fold_constant_convert_helper0<bfloat16>(constant, output_element_type);
+    case element::Type_t::f16:
+        return fold_constant_convert_helper0<float16>(constant, output_element_type);
+    case element::Type_t::f32:
+        return fold_constant_convert_helper0<float>(constant, output_element_type);
+    case element::Type_t::f64:
+        return fold_constant_convert_helper0<double>(constant, output_element_type);
+    case element::Type_t::i8:
+        return fold_constant_convert_helper0<int8_t>(constant, output_element_type);
+    case element::Type_t::i16:
+        return fold_constant_convert_helper0<int16_t>(constant, output_element_type);
+    case element::Type_t::i32:
+        return fold_constant_convert_helper0<int32_t>(constant, output_element_type);
+    case element::Type_t::i64:
+        return fold_constant_convert_helper0<int64_t>(constant, output_element_type);
+    case element::Type_t::u8:
+        return fold_constant_convert_helper0<uint8_t>(constant, output_element_type);
+    case element::Type_t::u16:
+        return fold_constant_convert_helper0<uint16_t>(constant, output_element_type);
+    case element::Type_t::u32:
+        return fold_constant_convert_helper0<uint32_t>(constant, output_element_type);
+    case element::Type_t::u64:
+        return fold_constant_convert_helper0<uint64_t>(constant, output_element_type);
+    }
+#if !(defined(__GNUC__) && (__GNUC__ == 4 && __GNUC_MINOR__ == 8))
+#pragma GCC diagnostic pop
+#endif
+}
+
+void pass::ConstantFolding::construct_constant_convert()
+{
+    auto constant_label = make_shared<pattern::op::Label>(
+        element::i32, Shape{2, 3, 4}, pattern::has_class<op::Constant>());
+    auto convert_op = make_shared<op::Convert>(constant_label, element::i64);
+
+    auto constant_convert_callback = [constant_label](pattern::Matcher& m) {
+        NGRAPH_DEBUG << "In callback for constant_convert_callback against node = "
+                     << m.get_match_root()->get_name();
+
+        auto pattern_map = m.get_pattern_map();
+
+        auto constant_match = static_pointer_cast<op::Constant>(pattern_map[constant_label]);
+        auto convert_match = static_pointer_cast<op::Convert>(m.get_match_root());
+
+        replace_node(
+            m.get_match_root(),
+            fold_constant_convert(constant_match, convert_match->get_output_element_type(0)));
+        return true;
+    };
+
+    auto convert_matcher =
+        make_shared<pattern::Matcher>(convert_op, "ConstantFolding.ConstantConvert");
+    this->add_matcher(
+        convert_matcher, constant_convert_callback, {PassProperty::CHANGE_DYNAMIC_STATE});
```

Comment:
Does this callback actually change dynamic state? It seems only converting a constant to a particular type?

Reply:
Good point! Will fix.

Reply:
As I think about it, it looks like the one in `DynElimination` for `Transpose` was not necessary either. It will replace the `Transpose` only under circumstances where the shape was already statically known. So I've removed that one. I've also removed the `CHANGE_DYNAMIC_STATE` for `ShapeOf`, because the shape-of-shape will be static in all cases where we replace. If you could double-check my reasoning I would be grateful :)